src/experimental/theoretic_information.cpp

```diff
@@ -0,0 +1,644 @@
+        m_cell_probabilities.insert(std::pair<std::vector<int>, std::vector<std::vector<float>>>(
+        {cell[0], cell[1]},
+        {{measurements[0], measurements[1], measurements[2]}}
+        ));
+        visited_cells[cell[0]][cell[1]] = true;
+    }
+    else
+    {
+        if(visited_cells[cell[0]][cell[1]] == true)
+        {
+            /*
+                Compare the unknown probability, the smallest it is the most information we will have
+                from the occupied or free state
+            */
+            int idx = it_cell->second.size() - 1;
+            if(measurements[2] < it_cell->second[idx][2])
+            {
+                // std::cout << "Replacing:" << it_cell->second[idx][0] << ", " << it_cell->second[idx][1] << ", " << it_cell->second[idx][2] << std::endl;
+                // std::cout << "With:" << measurements[0] << ", " << measurements[1] << ", " << measurements[2] << std::endl;
+
+                // Replacing
+                it_cell->second[idx][0] = measurements[0];
+                it_cell->second[idx][1] = measurements[1];
+                it_cell->second[idx][2] = measurements[2];
+            }
+        }
+        else
+        {
+            // Cell is already in the map, only add the next measurement outcome
+            it_cell->second.push_back({measurements[0], measurements[1], measurements[2]});
+            visited_cells[cell[0]][cell[1]] = true;
+        }
+    }
+}
+
+std::pair<std::vector<int>, std::vector<int>> TeorethicInformation::Bresenham(int x_1, int y_1, int x_2, int y_2)
+{
+    /*
+        To find the set of cells hit by a laser beam
+    */
+    std::vector<int> x_bres;
+    std::vector<int> y_bres;
+
+    int x = x_1;
+    int y = y_1;
+
+    int delta_x = abs(x_2 - x_1);
+    int delta_y = abs(y_2 - y_1);
+
+    int s_x = getSign(x_1, x_2);
+    int s_y = getSign(y_1, y_2);
+    bool interchange = false;
+
+    if (delta_y > delta_x)
+    {
+        int temp = delta_x;
+        delta_x = delta_y;
+        delta_y = temp;
+        interchange = true;
+    }
+    else
+    {
+        interchange = false;
+    }
+
+    int a_res = 2 * delta_y;
+    int b_res = 2 * (delta_y - delta_x);
+    int e_res = (2 * delta_y) - delta_x;
+
+    x_bres.push_back(x);
+    y_bres.push_back(y);
+
+    for (int i = 1; i < delta_x; ++i)
+    {
+        if (e_res < 0)
+        {
+            if (interchange)
+            {
+                y += s_y;
+            }
+            else
+            {
+                x += s_x;
+            }
+            e_res += a_res;
+        }
+        else
+        {
+            y += s_y;
+            x += s_x;
+            e_res += b_res;
+        }
+        x_bres.push_back(x);
+        y_bres.push_back(y);
+    }
+    return std::pair<std::vector<int>, std::vector<int>>{x_bres, y_bres};
+}
+
+float TeorethicInformation::calculateEntropy(float probability)
+{
+    /*
+        To calculate the entropy
+    */
+    return probability * log2(probability);
```

Comment:
@kmilo7204 meta: strictly speaking, this is not entropy but information content (see [definition](https://en.wikipedia.org/wiki/Entropy_(information_theory))). Also, this should be:
```suggestion
    return -probability * log2(probability);
```
Note the negative sign.